### PR TITLE
podvm changes: add tdx deps to Azure image and gRPC build of AA

### PR DESCRIPTION
**Pull request by jepio**
Add the option of including a gRPC attestation-agent in podvm image, and launch it using systemd. Use that in the Azure image.
Also install TDX deps to Azure image, so that AA can be built with all attesters.

### CODE REVIEW

File: azure/image/misc-settings.sh [hunk 1]
```diff
@@ -40,8 +40,12 @@ fi
 
 if [[ "$CLOUD_PROVIDER" == "azure" && "$PODVM_DISTRO" == "ubuntu" ]]; then
     export DEBIAN_FRONTEND=noninteractive
+    codename=$(. /etc/os-release; echo "$VERSION_CODENAME")
+    curl -fsSL https://download.01.org/intel-sgx/sgx_repo/ubuntu/intel-sgx-deb.key | sudo apt-key add -
+    echo "deb [arch=amd64] https://download.01.org/intel-sgx/sgx_repo/ubuntu ${codename} main" | sudo tee /etc/apt/sources.list.d/intel-sgx.list
```
Comment (katexochen) [inline azure/image/misc-settings.sh:45]:
```suggestion
    curl -fsSL https://download.01.org/intel-sgx/sgx_repo/ubuntu/intel-sgx-deb.key | sudo apt-key add -
    echo "deb [arch=amd64] https://download.01.org/intel-sgx/sgx_repo/ubuntu $(lsb_release -cs) main" | sudo tee /etc/apt/sources.list.d/intel-sgx.list
```

**Comment by jepio** [comment]
Let's not merge this yet, we had some issues with the test image (image pull and podvm start took longer than expected).

**Comment by surajssd** [comment]
@jepio what's the state of this PR? Should we move forward with it or just right now add the grpc based service?

**Comment by bpradipt** [comment]
TDX deps are already part of the image and with move to cdh, grpc build of AA is no longer needed.
Closing this PR.
@jepio please open it if you think otherwise.

**Pull request closed without merging**